gnutls: build Guile bindings

aconchillo opened the pull request:
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This takes a different approach than #59481 by simply building Guile bindings by default when installing `gnutls`. However, that means that `guile` is now required.

```
❯ brew audit --strict gnutls
gnutls:
  * Formula gnutls contains deprecated SPDX licenses: ["LGPL-2.1"].
```

gromgit commented:
`LGPL-2.1` is indeed a [deprecated license ID](https://spdx.org/licenses/), and it's not how GnuTLS is licensed anyway. Based on https://gitlab.com/gnutls/gnutls/-/blob/master/LICENSE, the correct stanza is more like:
```
license ["LGPL-2.1-or-later", "GPL-3.0-only"]
```

gromgit commented:
Sorry, that license syntax was for dual-licensing, which doesn't apply here. Use the following instead:
```
# license ""LGPL-2.1-or-later AND GPL-3.0-only" - review syntax after resolving https://github.com/Homebrew/brew/pull/8260
license "GPL-3.0-only"
```

aconchillo commented:
Thanks @gromgit. I will update once https://github.com/Homebrew/brew/pull/8260 is resolved.

I will also close #59481 in favor of this one since there's no way we can fix the dynamic library issue.

aconchillo commented:
For now updated it to use `license "LGPL-2.1-or-later"`.

gromgit commented:
Please use the exact `license` syntax that I mentioned. Because different parts of GnuTLS are licensed differently, Homebrew currently requires the most restrictive license be listed.

aconchillo commented:
@gromgit done! apologies, i misunderstood your previous message.

aconchillo commented:
hi @gromgit. is there any way for me to remove the `deprectaed license` label and re-run the checks. I believe they failed for some non-gnutls related issues.

SMillerDev commented:
You can rebase and push

aconchillo commented:
duh... true. 🤦‍♂️

aconchillo commented:
Any idea what's going on with this

```
Compiling Agda.Primitive in /usr/local/Cellar/agda/2.6.1/share/x86_64-osx-ghc-8.10.1/Agda-2.6.1/lib/prim/Agda/Primitive.agdai to /private/tmp/agda-test-20200814-1559-dm8svm/./jAgda.Agda.Primitive.js
Compiling SimpleTest in /private/tmp/agda-test-20200814-1559-dm8svm/SimpleTest.agdai to /private/tmp/agda-test-20200814-1559-dm8svm/./jAgda.SimpleTest.js
Error: agda: failed
An exception occurred within a child process:
  MethodDeprecatedError: Calling Language::Haskell::Cabal.cabal_sandbox is deprecated! There is no replacement.
Please report this issue to the homebrew/core tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/agda.rb:150
```

aconchillo commented:
Seems related to this issue https://github.com/Homebrew/homebrew-core/issues/58984

aconchillo commented:
This PR https://github.com/Homebrew/homebrew-core/pull/59709 will fix the issue once merged.

aconchillo commented:
Rebased now that https://github.com/Homebrew/homebrew-core/pull/59709 is merged and added a caveat I found was necessary to use the bindings properly.

SMillerDev reviewed (approved):
Thanks @aconchillo  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉
